libs/utils.py

```
@@ -0,0 +1,320 @@
+
+def validate_primary_key_column(df, type, prime_key):
+    """
+    This function is used to validate if primary key is present in dataframe & unique
+    :param df: Dataframe
+    :param type: source/target
+    :param prime_key: name of a column which acts like Primary key/composite in source/target file sheet
+    :return: True/None: True--> only if validation passed  & if validation failed exception will be thrown
+    """
+    try:
+        print(f"Validating the if Primary key is available in df..")
+        if not prime_key.upper() in [col.upper() for col in df.columns.to_list()]:
```

Ruff

Test for membership should be `not in`

Convert to `not in`

(E713)